data/transactions/logic/eval_test.go

```diff
@@ -1880,25 +1923,27 @@ func TestTxn(t *testing.T) {
 				clearProgramHash[:],
 			}
 			// Since we test GroupIndex ==3, we need a larger group
-			ep := defaultEvalParams(txn, txn, txn, txn)
-			ep.TxnGroup[2].EvalDelta.Logs = []string{"x", "prefilled"}
+			sep, aep := defaultEvalParams(txn, txn, txn, txn)
 			if v < txnEffectsVersion {
-				testLogicFull(t, ops.Program, 3, ep)
+				testLogicFull(t, ops.Program, 3, sep)
 			} else {
 				// Starting in txnEffectsVersion, there are fields we can't access in Logic mode
-				testLogicFull(t, ops.Program, 3, ep, "not allowed in current mode")
+				testLogicFull(t, ops.Program, 3, sep, "not allowed in current mode")
 				// And the early tests use "arg" a lot - not allowed in stateful. So remove those tests.
 				lastArg := strings.Index(source, "arg 10\n==\n&&")
 				require.NotEqual(t, -1, lastArg)
 
-				appSafe := "int 1" + strings.Replace(source[lastArg+12:], `txn Sender
+				if v >= appsEnabledVersion {
+					aep.TxnGroup[2].EvalDelta.Logs = []string{"x", "prefilled"}
```

Comment:
The guard isn't needed anymore. I originally added it because of the complicated way I was constructing a single `ep` that was supposed to be good for apps and sigs.  I added an app call in, so I would get a full `ep`, but only if the version >= appsEnabled (so I wouldn't create _other_ problems). But now that's simplified, so even though apps aren't enabled in a version, I create a usable `ep`.

In a way, it's nonsense - why test how apps work in a version before they are runnable?  But the extra check just complicates the flow, so I'll remove it.

Reply:
My question was more about the `aep.TxnGroup[2].EvalDelta.Logs = []string{"x", "prefilled"}` line

Reply:
Oh, I'll look more closely, but I assume that was so `gitxn Logs` could be tested.

But I'm finding that entire Replace suspicious. Digging in a bit.

Reply:
It's so `gtxn 2 NumLogs` works.
(not `gitxn` as there is no inner here)

And the Replace is fine. non sequitor.

Reply:
Added a small comment to explain.